tests/ieee802154_submac/main.c

```diff
@@ -0,0 +1,361 @@
+{
+    ieee802154_hal_test_init_devs();
+
+    netdev_t *dev = (netdev_t *)&netdev_submac;
+
+    dev->event_callback = _event_cb;
+    netdev_ieee802154_submac_init(&netdev_submac,
+                                  ieee802154_hal_test_get_dev(RADIO_DEFAULT_ID));
+    dev->driver->init(dev);
+    return 0;
+}
+
+uint8_t payload[] =
+    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam ornare lacinia mi elementum interdum ligula.";
+
+static iolist_t iol_hdr;
+
+static int send(uint8_t *dst, size_t dst_len,
+                size_t len)
+{
+    uint8_t flags;
+    uint8_t mhr[IEEE802154_MAX_HDR_LEN];
+    int mhr_len;
+
+    le_uint16_t src_pan, dst_pan;
+    iolist_t iol_data = {
+        .iol_base = payload,
+        .iol_len = len,
+        .iol_next = NULL,
+    };
+
+    flags = IEEE802154_FCF_TYPE_DATA | 0x20;
+    src_pan = byteorder_btols(byteorder_htons(0x23));
+    dst_pan = byteorder_btols(byteorder_htons(0x23));
+    uint8_t src_len = 8;
+    void *src = &netdev_submac.submac.ext_addr;
+
+    /* fill MAC header, seq should be set by device */
+    if ((mhr_len = ieee802154_set_frame_hdr(mhr, src, src_len,
+                                            dst, dst_len,
+                                            src_pan, dst_pan,
+                                            flags, seq++)) < 0) {
+        puts("txtsnd: Error preperaring frame");
+        return 1;
+    }
+
+    iol_hdr.iol_next = &iol_data;
+    iol_hdr.iol_base = mhr;
+    iol_hdr.iol_len = mhr_len;
+
+    netdev_t *dev = (netdev_t *)&netdev_submac;
+
+    dev->driver->send(dev, &iol_hdr);
+    return 0;
+}
+
+static inline int _dehex(char c, int default_)
+{
+    if ('0' <= c && c <= '9') {
+        return c - '0';
+    }
+    else if ('A' <= c && c <= 'F') {
+        return c - 'A' + 10;
+    }
+    else if ('a' <= c && c <= 'f') {
+        return c - 'a' + 10;
+    }
+    else {
+        return default_;
+    }
+}
+
+static size_t _parse_addr(uint8_t *out, size_t out_len, const char *in)
+{
+    const char *end_str = in;
+    uint8_t *out_end = out;
+    size_t count = 0;
+    int assert_cell = 1;
+
+    if (!in || !*in) {
+        return 0;
+    }
+    while (end_str[1]) {
+        ++end_str;
+    }
+
+    while (end_str >= in) {
+        int a = 0, b = _dehex(*end_str--, -1);
+        if (b < 0) {
+            if (assert_cell) {
+                return 0;
+            }
+            else {
+                assert_cell = 1;
+                continue;
+            }
+        }
+        assert_cell = 0;
+
+        if (end_str >= in) {
+            a = _dehex(*end_str--, 0);
+        }
+
+        if (++count > out_len) {
+            return 0;
+        }
+        *out_end++ = (a << 4) | b;
+    }
+    if (assert_cell) {
+        return 0;
+    }
+    /* out is reversed */
+
+    while (out < --out_end) {
+        uint8_t tmp = *out_end;
+        *out_end = *out;
+        *out++ = tmp;
+    }
+
+    return count;
+}
+
+int txtsnd(int argc, char **argv)
```

Comment:
Isn't there a `gnrc_txtsnd` already?

Reply:
yes, but this test has no GNRC dependencies, so we couldn't use that one :(

Reply:
Maybe it can be moved elsewhere to be put in common?

Reply:
we could do that. Ideally it would be cool to have an L2 interface here (this is exactly what `txtsnd` does)